### PR TITLE
Handle labels

### PR DESCRIPTION
**Pull request by oerc0122**
- Add function to turn castep labels into `dict`s
- Add function to turn `str`ings/`dict`s into `NamedTuples` (in new `processing.py`)
- Add parsing for labels on Born Charges
- Add test for the above

Addresses #104

### CODE REVIEW

**Comment by jkshenton** [comment]
Thanks for this! 

I've added one comment - about standardising to the  (Sp:tag [label])  form. 

The only other thing I noticed was the output to json (possibly other formats?) makes these keys look like this:
`"H [H1]_1"` which could be a bit clunky later on (?) when dealing with these json files. Is there a better alternative, I wonder? (I guess this is handled in the `json_safe` function?).

File: castep_outputs/test/test_castep_parser.py [hunk 1]
```diff
@@ -284,9 +284,10 @@ def test_get_atom_struct_labelled(self):
 
         test_dict = parse_castep_file(test_text)[0]
 
-        self.assertEqual(test_dict, {'initial_positions': {
-            ('H [H1]', 1): (0.0, 0.0, 0.0),
-            ('H [H2]', 2): (-0.0, -0.0, 0.166667)}
+        self.assertEqual(test_dict, {'initial_positions': {('H', 1): (0.0, 0.0, 0.0),
```
[the remark this answers is not in the document]
Comment (oerc0122) [reply castep_outputs/test/test_castep_parser.py]: Good spot. I had forgotten I'd handled the labels differently on `ionic_positions`. Should now be better. We were almost left with a hybrid of the worst of all worlds. Also need to fix the handling of an empty label which is written as `"       "`, i.e. my RE picks up the `x` delimiting the table.

**Comment by oerc0122** [comment]
> Thanks for this!
> 
> I've added one comment - about standardising to the (Sp:tag [label]) form.
> 
> The only other thing I noticed was the output to json (possibly other formats?) makes these keys look like this: `"H [H1]_1"` which could be a bit clunky later on (?) when dealing with these json files. Is there a better alternative, I wonder? (I guess this is handled in the `json_safe` function?).

There're a few approaches we could take to this. 

1. Strip the label notation out before dumping to json/yaml, which would be fine as there'd be the label mapping block, but may be quite clunky. 
2. Separate the label by an underscore rather than a space
3. Remove the space leaving ("H:tag[label]")
4. Leave it as-is

Most of these affect the recovery of the original data from the json negatively. In general, I think leaving as-is is the best option as this leaves the user the ability to quickly run a "split on `_`" and parse the second element as an `int`, and recover the original data. The problem is that json doesn't permit tuples as indices for a dict. The `_` is just a quick, reversible way around that.

**Review by jkshenton — APPROVED** [review]
Thanks - that looks good to me! 

For the json output - I guess I was worried if anyone used `_` in their labels, but I guess even in that case, splitting on the final `_` only would still allow the best path to recovery. 

Cheers

**Pull request merged**